src/huggingface_hub/inference/_client.py

```diff
@@ -860,10 +860,10 @@ def chat_completion(
                     stream=stream,
                 )
             except HTTPError as e:
-                if e.response.status_code in (400, 404, 500):
+                if e.response.status_code in (400, 500):
```

Comment:
There should really be only 1 error code here.
Can you check on a real deployment or in TGI source code what happens on non existing template models ?

I think it should be a 4xx since it's not really a server error.  No idea for which one is more appropriate.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses

Reply:
> Can you check on a real deployment or in TGI source code what happens on non existing template models ?

It returns a 
```
huggingface_hub.utils._errors.HfHubHTTPError: 422 Client Error: Unprocessable Entity for url: https://api-inference.huggingface.co/models/gpt2/v1/chat/completions (Request ID: 3Y0mKxT7AmdMSZsfNjcQA)

Template error: template not found
```

But in fact this is not even the problem. Since client-side rendering has been completely removed (in https://github.com/huggingface/huggingface_hub/pull/2258), there is no point treating TGI-served and other models differently. I've updated the PR to simplify the logic which will finally avoid hiding errors to the user.